### DISCUSSION TITLE
Slither cannot run because of the incompatible solidity versions of PuppyRaffle.sol and its dependencies

### DISCUSSION THREAD

**Discussion by fabriziogianni7** [Cyfrin/security-and-auditing-full-course-s23#155] [General]
# Description

![Schermata 2024-04-12 alle 10 41 21](https://github.com/Cyfrin/security-and-auditing-full-course-s23/assets/46995085/ecda236a-628b-4d50-842c-471cb8a28783)

`slither .` is failing because it encounters incompatible versions of solidity in PuppyRaffle and its dependencies.
I can't find how to fix it here without setting a higher version on the contract 🤔🤔🤔

**Comment by fabriziogianni7** [comment]
temporary fix:
manually updating the version of PuppyRaffle to 0.8.25 and then running
`slither src --ignore-compile --solc-remaps @openzeppelin=lib/openzeppelin-contracts --exclude-dependencies`

I also found that the version 3.4.0 of openzeppelin, in ERC721 contract, does not have _exists() and totalSupply(). so I just commented it.

**Reply by EngrPips** [reply]
Awesome, Thank you so much for posting this as it would help anyone that run into similar issue
